libcontainer/container_linux.go

```diff
@@ -244,6 +248,9 @@ func (c *linuxContainer) Set(config configs.Config) error {
 			return err
 		}
 	}
+	// reset SkipDevices before saving the state as this will cause major issues when
+	// trying to exec into the container or other process related actions.
+	config.Cgroups.SkipDevices = false
```

Comment:
You're not resetting this on error.

I think, if it has to be done this way, using a `defer` will be more shatterproof:

```go
	config.Cgroups.SkipDevices = true
	defer func() {config.Cgroups.SkipDevices = false}
```

Reply:
Oops, I was wrong about `defer` 😊